sdk/search/search-documents/src/searchIndexingBufferedSender.ts

```diff
@@ -377,9 +386,35 @@ export class SearchIndexingBufferedSender<T> {
       this.batchObject = new IndexDocumentsBatch<T>();
       while (actions.length > 0) {
         const actionsToSend = actions.splice(0, this.initialBatchActionCount);
-        await this.submitDocuments(actionsToSend, options);
+        await this.submitDocuments(this.pruneActions(actionsToSend, actions), options);
+      }
+    }
+  }
+
+  private pruneActions(
+    batch: IndexDocumentsAction<T>[],
+    actions: IndexDocumentsAction<T>[]
+  ): IndexDocumentsAction<T>[] {
+    let counter: number = 0;
+    const hashSet: Set<string> = new Set<string>();
+    const resultBatch: IndexDocumentsAction<T>[] = [];
+
+    while (counter < batch.length) {
+      const { __actionType, ...document } = batch[counter];
```

Comment:
we don't need to do this part right? Since IndexDoucmentsAction<T> is a superset of T, we should be able to pass it directly to documentKeyRetriever.

Reply:
Yes. Removed it.